pkg/ingester/ingester.go

```diff
@@ -2440,6 +2457,31 @@ func (i *Ingester) getInstanceLimits() *InstanceLimits {
 	return l
 }
 
+// PrepareShutdownHandler changes the configuration of the ingester such that when
+// it is stopped, it will:
+//   - Change the state of ring to stop accepting writes.
+//   - Flush all the chunks to long-term storage.
+//
+// It also creates a file on disk which is used to re-apply the configuration if the
+// ingester crashes and restarts before being permanently shutdown.
+func (i *Ingester) PrepareShutdownHandler(w http.ResponseWriter, r *http.Request) {
+	if err := i.shutdownMarker.Create(); err != nil {
```

Comment:
I'd suggest to change the endpoint slightly:

- on GET, return whether marker exists
- on POST, create marker

Reply:
Done. Also added support for `DELETE` to unset the marker like @colega mentioned